dev/tests/integration/testsuite/Magento/Catalog/Controller/Adminhtml/CategoryTest.php

```diff
@@ -890,4 +890,70 @@ public function testSaveWithCustomBackendNameAction(): void
             MessageInterface::TYPE_ERROR
         );
     }
+
+    /**
+     * Verify that the category cannot be saved if category name can not be converted to Latin (like Thai)
+     *
+     * @return void
+     */
+    public function testSaveWithThaiCategoryNameAction(): void
+    {
+        $categoryName = 'ประเภท';
+        $errorMessage = 'Invalid URL key. The "%1" category name can not be used to generate Latin URL key. ' .
+            'Please add URL key or change category name using Latin letters and numbers to avoid generating ' .
+            'URL key issues.';
+        $inputData = [
+            'name' => $categoryName,
+            'use_config' => [
+                'available_sort_by' => 1,
+                'default_sort_by' => 1
+            ],
+            'is_active' => '1',
+            'include_in_menu' => '1',
+        ];
+        $this->getRequest()->setMethod(HttpRequest::METHOD_POST);
+        $this->getRequest()->setPostValue($inputData);
+        $this->dispatch('backend/catalog/category/save');
+        $this->assertSessionMessages(
+            $this->equalTo(
+                [
+                    (string)__($errorMessage, $categoryName)
+                ]
+            ),
+            MessageInterface::TYPE_ERROR
+        );
+    }
+
+    /**
+     * Verify that the category cannot be saved if category URL key can not be converted to Latin (like Thai)
+     *
+     * @return void
+     */
+    public function testSaveWithThaiCategoryUrlKeyAction(): void
+    {
+        $categoryPath = 'ประเภท';
```

Comment:
That’s not actually category path. So would be great to rename it 
```suggestion
        $categoryUrlKey = 'ประเภท';
```

Reply:
Nice catch! Changed, thank you.